packages/nimble-components/src/table/types.ts

```diff
@@ -0,0 +1,14 @@
+// TableDataValue describes the supported value types for values
+// within the data given to a table.
+export type TableDataValue =
```

Comment:
Maybe we can use the name `TableCellData`, I think that would compliment the name `TableRowData` better

We should also align on one of the names 'data' vs 'value', seems like we are aligning on data.

Reply:
`TableCellData` is the proposed name of the interface that represents the data passed into a cell in #900. I'm not opposed to changing this to `TableCellData`, but it just means we'll need to come up with a different name for that interface.

FYI @atmgrifter00

Reply:
I agree with @mollykreis that `TableCellData` seems more appropriate for its currently proposed usage. While I agree that `TableDataValue` may not be immediately understandable for its purpose I think it's closer than `TableCellData`, as that term suggests the possibility of a set of values (which is appropriate for a table cell).

Reply:
Part of the issue here (at least for me) is that we don't have a good term to refer to the key/value pairs for a row. Calling the key/value pairs "cells" or "cell data" is a bit misleading since a cell is something in the table that might use a combination of those key/value pairs. However, I'm not sure what to call them other than a "key/value pair of data associated with a row". 

That being said some alternative suggestions for this type could be:
- `TableData` - This is what @atmgrifter00 used in #900 to refer to this type, though I'm not a huge fan of it because in my mind `TableData` should be of type `TableRowData[]`
- `TableRowDataValue`
- `TableRowDataType`
- `TableRowValueType`
- `TableRowValueData`

I'm not convinced that any of the options I mentioned are better than what is currently in the code, but I'm curious if anyone has a differing opinion, or if these suggestions spark other ideas for anyone.

Reply:
Another option:
- `TableRowEntryData` -- pulling `Entry` here from the javascript concept of `Object.entries`, so maybe a "row entry" could refer to one of the key/value pairs within a row

Reply:
The decision in an offline discussion is to use the following terminology to help remove ambiguity between types associated with the data and types associated with elements:
- `interface TableRecord` - The data associated with a row in the table
- `type TableFieldName` - The type requirements of keys within `TableRecord`
- `type TableFieldValue` - The type requirements of values within `TableRecord`

I will update the code accordingly